cachito/workers/tasks/gomod.py

```diff
@@ -180,30 +184,33 @@ def fetch_gomod_source(request_id, dep_replacements=None, package_configs=None):
         log.info(
             "Fetching the gomod dependencies for request %d in subpath %s", request_id, subpath
         )
-        set_request_state(
-            request_id,
-            "in_progress",
-            f'Fetching the gomod dependencies at the "{subpath}" directory',
-        )
-        request = get_request(request_id)
-        gomod_source_path = str(bundle_dir.app_subpath(subpath).source_dir)
-        try:
-            gomod = resolve_gomod(
-                gomod_source_path, request, dep_replacements, bundle_dir.source_dir
+        with tracer.start_as_current_span("fetching gomod deps in " + subpath):
```

Comment:
Maybe it's enough to have a span in the `resolve_gomod` file, as it is the function that is doing the bulk of the work.